### PR TITLE
keep in holdover until locked-holdover-acquired

### PR DESCRIPTION
**Pull request by vitus133**
https://github.com/k8snetworkplumbingwg/ptp-operator/issues/9

Simplify DPLL eventing logic by decoupling holdover and LHAQ states.
Remove "locked" DPLL state handling, keeping in Holdover until the LHAQ state reached.
Fix a corner case when holdover timer keeps running when DPLL goes to unlocked
Fix a race condition in tests where event source was not fast enough to be set before the event processing 

/cc @aneeshkp @josephdrichard

### CODE REVIEW

File: pkg/dpll/dpll.go [hunk 1]
```diff
@@ -627,47 +628,62 @@ func (d *DpllConfig) MonitorDpll() {
...
 	dpllStatus := d.getWorseState(d.phaseStatus, d.frequencyStatus)
+
 	switch dpllStatus {
 	case DPLL_FREERUN, DPLL_INVALID, DPLL_UNKNOWN:
 		d.inSpec = false
 		d.sourceLost = true
-		if d.hasGNSSAsSource() && d.onHoldover {
-			d.holdoverCloseCh <- true
+		glog.Infof("on holdover %t has GNSS source %t", d.onHoldover, d.hasGNSSAsSource())
+		if d.onHoldover {
```
Comment (aneeshkp) [inline pkg/dpll/dpll.go:637]: we need to gate this for d.hasGNSSAsSource() ; if  this clock id has GNSS only then we control holdover

Comment (aneeshkp) [reply pkg/dpll/dpll.go:637]: we don't close holdover once its in holdover if other nic state change to Freerun etc

File: pkg/dpll/dpll.go [hunk 2]
```diff
@@ -677,17 +693,10 @@ func (d *DpllConfig) stateDecision() {
...
-				d.state = event.PTP_HOLDOVER
-				go d.holdover()
-			}
-			return // do not send event holdover  will handle it
-		case !d.inSpec: // this is for GNSS only
-			glog.Infof("dpll is not in spec, state is DPLL_LOCKED_HO_ACQ or DPLL_HOLDOVER, offset is out of range, state is FREERUN(%s)", d.iface)
+		} else {
+			glog.Infof("dpll is not in spec, state is DPLL_LOCKED_HO_ACQ, offset is out of range, state is FREERUN(%s)", d.iface)
 			d.state = event.PTP_FREERUN
+			d.inSpec = true
```
Comment (aneeshkp) [inline pkg/dpll/dpll.go:699]: if state is freeRUN , how is  inSpec is true

File: pkg/dpll/dpll.go (hunk 1 above)
Comment (vitus133) [reply pkg/dpll/dpll.go]: Fixed, thanks!

File: pkg/dpll/dpll.go (hunk 2 above)
Comment (vitus133) [reply pkg/dpll/dpll.go]: Indeed, fixed

**Review by aneeshkp — APPROVED** [review]
/lgtm
